clients/common/nm-vpn-helpers.c

```diff
@@ -247,3 +253,522 @@ nm_vpn_openconnect_authenticate_helper (const char *host,
+				if (g_ascii_isspace (ch)) {
+					/* wg-setconf strips all whitespace before parsing the content. That means,
+					 * *[I nterface]" will be accepted. We do that too. */
+					continue;
+				}
+				if (ch == '#')
+					break;
+				line[n++] = line[l];
+			}
+			if (n == 0)
+				continue;
+			line[n] = '\0';
+		}
+
+		if (g_ascii_strcasecmp (line, "[Interface]") == 0) {
+			if (!_wg_complete_peer (&data_peers,
+			                        g_steal_pointer (&current_peer),
+			                        current_peer_start_line_nr,
+			                        filename,
+			                        error))
+				return FALSE;
+			line_context = LINE_CONTEXT_INTERFACE;
+			continue;
+		}
+
+		if (g_ascii_strcasecmp (line, "[Peer]") == 0) {
+			if (!_wg_complete_peer (&data_peers,
+			                        g_steal_pointer (&current_peer),
+			                        current_peer_start_line_nr,
+			                        filename,
+			                        error))
+				return FALSE;
+			current_peer_start_line_nr = line_nr;
+			current_peer = nm_wireguard_peer_new ();
+			line_context = LINE_CONTEXT_PEER;
+			continue;
+		}
+
+		if (line_context == LINE_CONTEXT_INTERFACE) {
+
+			if (line_match (line, "Address", &matched_key, &value)) {
+				GPtrArray **p_data_addr;
+				NMIPAddr addr_bin;
+				int addr_family;
+				int prefix_len;
+
+				if (!nm_utils_parse_inaddr_prefix_bin (AF_UNSPEC,
+				                                       value,
+				                                       &addr_family,
+				                                       &addr_bin,
+				                                       &prefix_len))
+					goto fail_invalid_value;
+
+				p_data_addr =   (addr_family == AF_INET)
+				              ? &data_addr_v4
+				              : &data_addr_v6;
+
+				if (!*p_data_addr)
+					*p_data_addr = g_ptr_array_new_with_free_func ((GDestroyNotify) nm_ip_address_unref);
+
+				g_ptr_array_add (*p_data_addr,
+				                 nm_ip_address_new_binary (addr_family,
+				                                           &addr_bin,
+				                                             prefix_len == -1
+				                                           ? ((addr_family == AF_INET) ? 32 : 128)
+				                                           : prefix_len,
+				                                           NULL));
+				continue;
+			}
+
+			if (line_match (line, "MTU", &matched_key, &value)) {
+				i64 = _nm_utils_ascii_str_to_int64 (value, 0, 0, G_MAXUINT32, -1);
+				if (i64 == -1)
+					goto fail_invalid_value;
+
+				/* TODO: not yet implemented. Ignore the value.
+				 *
+				 * wg-quick accepts the "MTU" value, but it also fetches routes to
+				 * autodetect it. */
+				continue;
+			}
+
+			if (line_match (line, "DNS", &matched_key, &value)) {
+				char addr_s[NM_CONST_MAX (INET_ADDRSTRLEN, INET6_ADDRSTRLEN)];
+				GPtrArray **p_data_dns;
+				NMIPAddr addr_bin;
+				int addr_family;
+
+				if (!nm_utils_parse_inaddr_bin (AF_UNSPEC,
```

Comment:
Note that fields like Address=, DNS=, and AllowedIPs= can take multiple entries on a single line, separated by a comma.

Reply:
fixed